server/connectors/api-connector-jdbc/src/main/scala/com/prisma/api/connector/jdbc/database/FilterConditionBuilder.scala

```diff
@@ -54,34 +73,61 @@ trait FilterConditionBuilder extends BuilderBase {
     }
   }
 
-  private def relationFilterStatement(alias: String, relationFilter: RelationFilter): Condition = {
-    val relationField         = relationFilter.field
-    val relation              = relationField.relation
-    val newAlias              = relationField.relatedModel_!.dbName + "_" + alias
-    val nestedFilterStatement = buildConditionForFilter(relationFilter.nestedFilter, newAlias)
+  private def relationFilterStatement(alias: String, relationFilter: RelationFilter, relField: Option[Field[AnyRef]], invert: Boolean): Condition = {
+    // this skips intermediate table when there is no condition on it
+    // artists(where:{albums_some:{tracks_some:{condition}}})
+    // albums(where: {Tracks_some:{ MediaType:{Name_starts_with:""}, Genre:{Name_starts_with:""}}})
+    // this will be an andFilter around the two nested ones and will not be skipped at the moment
+    // implicit AND like above as well as explicit AND, OR, NOT can be improve
```

Comment:
the last sentence in the comment does not make sense.

Reply:
made the comment more clear